app/lib/search/backend.dart

```diff
@@ -154,19 +156,25 @@ class SearchBackend {
       likeCount: p.likes,
       grantedPoints: analysisView.report?.grantedPoints ?? 0,
       maxPoints: analysisView.report?.maxPoints ?? 0,
-      dependencies: _buildDependencies(analysisView),
+      dependencies: _buildDependencies(pv.pubspec, analysisView),
       publisherId: p.publisherId,
       uploaderEmails: await _buildUploaderEmails(p),
       apiDocPages: apiDocPages,
       timestamp: DateTime.now().toUtc(),
     );
   }
 
-  Map<String, String> _buildDependencies(AnalysisView view) {
+  Map<String, String> _buildDependencies(Pubspec pubspec, AnalysisView view) {
     final Map<String, String> dependencies = <String, String>{};
     view.allDependencies?.forEach((pd) {
       dependencies[pd.package] = pd.dependencyType;
     });
+    pubspec.dependencies.forEach((package) {
+      dependencies[package] = DependencyTypes.direct;
+    });
+    pubspec.devDependencies.forEach((package) {
+      dependencies[package] = DependencyTypes.dev;
+    });
```

Comment:
Might you not want to re-order these? in case a package is both a dependency and a dev_dependency?